core-plugins/src/test/java/io/cdap/plugin/batch/joiner/JoinerTestRun.java

```diff
@@ -77,37 +82,55 @@ public void testInnerJoin() throws Exception {
      *
      */
 
+    ETLBatchConfig config = ETLBatchConfig.builder("* * * * *")
+      .addStage(filmStage)
+      .addStage(filmActorStage)
+      .addStage(filmCategoryStage)
+      .addStage(joinStage)
+      .addStage(joinSinkStage)
+      .addConnection(filmStage.getName(), joinStage.getName())
+      .addConnection(filmActorStage.getName(), joinStage.getName())
+      .addConnection(filmCategoryStage.getName(), joinStage.getName())
+      .addConnection(joinStage.getName(), joinSinkStage.getName())
+      .build();
+    ApplicationManager appManager = deployETL(config, "joiner-test");
```

Comment:
is it an issue if the same name is used across runs?

Reply:
Yes, I see this error when dataset name is reused:
```
Exception in thread "WorkflowDriver" io.cdap.cdap.api.dataset.lib.PartitionAlreadyExistsException: Dataset 'innerjoin-output' already has a partition with the same key: {year=2020, month=3, day=2, hour=14, minute=17}
```

Reply:
yes, so is it a problem if this same 'joiner-test' app name is used? does this need to be unique?

Reply:
The test passes when the 'joiner-test' app name is reused. Is it incorrect reuse the app name? If yes, I can pass in app name as param to `joinHelper`

Reply:
it's probably safer to generate UUID.

we've had bugs in the past where a test will start an app, then wait for at least one completed run. However, since the app name is re-used, and the app was started in a different test, it immediately finds a completed run and doesn't actually wait for the current one to complete.

Reply:
thanks, fixed to use UUID.